src/main/java/com/minecolonies/colony/permissions/Permissions.java

```diff
@@ -50,6 +52,10 @@
     private boolean            dirty       = false;
 
     /**
+     * The name of the owner.
+     */
+    private String ownerName ="";
```

Comment:
Instead of saving the owner name, we should save the UUID, then getting the name is a simple O(1) lookup, `players.get(ownerUUID).getName()`